src/lib/phdi-building-blocks/phdi_building_blocks/utils.py

```diff
@@ -170,10 +174,33 @@ def http_request_with_retry(
             return
     elif request_type == "GET":
         try:
-            response = requests.get(
+            response = http.get(
                 url,
                 headers=headers,
             )
-            return response
         except Exception:
             logging.exception("GET request to " + url + " failed.")
+
+    if (
+        cred_manager
+        and response.status_code == 401
+        and headers.get("Authorization", "").startswith("Bearer ")
+    ):
+        new_access_token = cred_manager.get_access_token().token
+
+        # New headers dict mostly for unit testing so we can test the header on the
+        # first and second attempt
+        reauth_headers = headers.copy()
+        reauth_headers["Authorization"] = f"Bearer {new_access_token}"
```

Comment:
I def get how this simplifies things for unit testing, but if that's the only real reason it's here is there a way we can move this to the actual testing file rather than be held in the function invocation? I know the performance hit of maintaining one extra dict for a call is negligible, I just hate the idea that we're carrying around an extra data structure just for some test cases.